File: Allegro5 Test/Floor.cpp
#include "Floor.h"
Floor::Floor(){}

void Floor::Destroy(){
	GameObject::Destroy();
}

void Floor::Init(ALLEGRO_BITMAP *image = 0, float x = 0, float y = 0, int boundX = 100, int boundY = 100){
	GameObject::Init(x, y ,0,0,0,0,boundX + 4, boundY);
	SetID(FLOOR);
	SetAlive(true);
	SetCollidable(true);
	SetCollideType(WHITE);

	frameWidth = boundX*2;
	frameHeight = boundY*2;

	if(image != 0){
		Floor::image = image;
	}
}

void Floor::Render(){
	GameObject::Render();

	al_draw_bitmap_region(image, x-(frameWidth/2), y-(frameHeight/2), frameWidth, frameHeight, x-(frameWidth/2), y-(frameHeight/2), 0);
}

void Floor::Update(){
	GameObject::Update();
}

bool Floor::CheckCollisions(GameObject *otherObject){
	return GameObject::CheckCollisions(otherObject);
}

void Floor::Collided(int objectID){
}

File: js/floor.js
// Floor class - base class for different floor types

class Floor extends GameObject {
    constructor() {
        super();
        this.setId(ID.FLOOR);
        this.setCollidable(true);
        this.setCollideType(COLLIDE_TYPE.WHITE);
    }
    
    init(texture, x, y, boundX, boundY) {
        super.init(x, y, 0, 0, 0, 0, boundX, boundY);
        
        if (texture) {
            this.sprite = new PIXI.Sprite(texture);
            this.sprite.x = x;
            this.sprite.y = y;
            this.sprite.width = boundX * 2;
            this.sprite.height = boundY * 2;
            this.sprite.anchor.set(0.5, 0.5);
        }
    }
    
    update() {
        // Floor objects don't need to update
    }
    
    render() {
        super.render();
    }
    
    checkCollisions(otherObject) {
        if (!this.collidable() || !otherObject.collidable()) {
            return false;
        }
        
        // Check if collision types are compatible
        if (this.getCollideType() !== otherObject.getCollideType()) {
            return false;
        }
        
        return super.checkCollisions(otherObject);
    }
    
    collided(otherObject) {
        // Floor collision handling - to be implemented based on game mechanics
    }
}